server/src/main/java/thesistrack/ls1/controller/ThesisController.java

```diff
@@ -0,0 +1,132 @@
+package thesistrack.ls1.controller;
+
+import com.nimbusds.jose.util.Resource;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.data.domain.Page;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import thesistrack.ls1.dto.PageResponse;
+import thesistrack.ls1.dto.ThesisDto;
+import thesistrack.ls1.service.ThesisService;
+
+import java.util.UUID;
+
+@Slf4j
+@RestController
+@RequestMapping("/v1/theses")
```

Comment:
Right now there is only v2 for applications as the only legacy endpoint is to create a application because in the future you will need to be authenticated for that. All other endpoints use v1 at the moment as there are no other legacy endpoints. Do you think all my endpoints should start with v2 except for the one legacy endpoint?

Reply:
I think all new endpoints / controllers and those relying on "new" DTOs should use a `/v2` path for consistency. Semantically, I'd expect v1 endpoints to be older/deprecated when v2 endpoints are available

Reply:
I see arguements for both sides but you convinced me, so I will change it